src/odemis/acq/move.py

```diff
@@ -1108,12 +1135,22 @@ def _transformFromMeteorToSEM(self, pos: Dict[str, float]) -> Dict[str, float]:
         md_calib = stage_md[model.MD_CALIB]
         sem_pos_active = stage_md[model.MD_FAV_SEM_POS_ACTIVE]
 
+        if self.fm_sample_z_starting_point is not None:
+            # transformed_pos -> sample_stage
+            sample_stage_pos = self.to_sample_stage_from_stage_position(transformed_pos, posture=FM_IMAGING)
+            # sample_stage_pos[z] override
+            sample_stage_pos["z"] = self.fm_sample_z_starting_point
+            # overridden sample_stage_pos -> stage bare
+            transformed_pos = self.from_sample_stage_to_stage_position(sample_stage_pos, posture=FM_IMAGING)
+            # Unset the snapshotted position
+            self.fm_sample_z_starting_point = None
```

Comment:
As mentioned in my previous comment, this part should be in in `getTargetPosition()`, when current posture is FM, and target is SEM.